bridge_adaptivity/module/views.py

```diff
@@ -183,6 +239,22 @@ class CollectionList(BaseCollectionView, ListView):
     context_object_name = 'collections'
     filter = 'collection_slug'
 
+    def get_context_data(self):
+        context = super().get_context_data()
+        # Get Module Groups where collections are used.
+        mg = ModuleGroup.objects.filter(collections__in=list(context['object_list'])).distinct()
+        # The "name" and "slug" are ModuleGroup fields
+        res = mg.values('name', 'slug', 'collections__slug').filter(
```

Comment:
module groups related to the user through the owner field and contributor field your current implementation does not look efficient enough. We can discuss it locally.

Reply:
I added Notes to fix it in the future.